metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_utils.py

```diff
@@ -59,6 +62,44 @@ class SnowflakeCommonMixin:
 
     platform = "snowflake"
 
+    @staticmethod
+    @lru_cache(maxsize=128)
+    def create_snowsight_base_url(account_id: str) -> Optional[str]:
+        cloud: Optional[str] = None
+        account_locator: Optional[str] = None
+        cloud_region_id: Optional[str] = None
+
+        if "." not in account_id:  # e.g. xy12345
+            account_locator = account_id.lower()
+            cloud_region_id = "us-west-2"
+        else:
+            parts = account_id.split(".")
+            if len(parts) == 2:  # e.g. xy12345.us-east-1
+                account_locator = parts[0].lower()
+                cloud_region_id = parts[1].lower()
+            elif len(parts) == 3 and cloud in ("aws", "gcp", "azure"):
+                # e.g. xy12345.ap-south-1.aws or xy12345.us-central1.gcp or xy12345.west-us-2.azure
+                # NOT xy12345.us-west-2.privatelink or xy12345.eu-central-1.privatelink
```

Comment:
Why not check that if it ends with "privatelink", and then automatically generate the snowsight URL from this?

Basically if the len(parts) == 3 and parts[2] == 'privatelink':
   # Generate the private link URL as 
   {account_locator}.{region}.privatelink.snowflakecomputing....

In fact, we should support this in general when connecting to Snowflake for Metadata Extraction (my guess is that we already do!)